android-app/src/main/java/org/example/app/view/NewsActivity.kt

```diff
@@ -6,26 +6,31 @@ package org.example.app.view
 
 import android.os.Bundle
 import androidx.lifecycle.ViewModelProvider
+import dagger.hilt.android.AndroidEntryPoint
 import dev.icerock.moko.mvvm.MvvmActivity
 import dev.icerock.moko.mvvm.createViewModelFactory
-import org.example.app.AppComponent
+import javax.inject.Inject
 import org.example.app.BR
 import org.example.app.R
 import org.example.app.databinding.ActivityNewsBinding
+import org.example.library.SharedFactory
 import org.example.library.domain.entity.News
 import org.example.library.feature.list.presentation.ListViewModel
 
 // MvvmActivity for simplify creation of MVVM screen with https://github.com/icerockdev/moko-mvvm
+@AndroidEntryPoint
 class NewsActivity : MvvmActivity<ActivityNewsBinding, ListViewModel<News>>() {
     override val layoutId: Int = R.layout.activity_news
     @Suppress("UNCHECKED_CAST")
     override val viewModelClass = ListViewModel::class.java as Class<ListViewModel<News>>
     override val viewModelVariableId: Int = BR.viewModel
 
+    @Inject lateinit var factory: SharedFactory
```

Comment:
i think will be better if we will inject here not `SharedFactory` but `NewsFactory`

Reply:
done